apps/web/src/components/Button/ButtonWithLinkAndEventLogging.tsx

```diff
@@ -0,0 +1,37 @@
+'use client';
+
+import { useCallback } from 'react';
+
+import logEvent, { AnalyticsEventData, CCAEventData } from 'libs/base-ui/utils/logEvent';
+
+import { ButtonWithLink, ButtonWithLinkProps } from './ButtonWithLink';
+
+export function ButtonWithLinkAndEventLogging({
+  eventName,
+  eventContext,
+  eventData,
+  ...buttonWithLinkProps
+}: ButtonWithLinkAndEventLogProps) {
+  const event: AnalyticsEventData = {
```

Comment:
you could wrap

`apps/web/app/(base-org)/layout.tsx`
 
with `<AnalyticsProvider context="base-org">`

and get contextually analytics for free with

`const { logEventWithContext } = useAnalytics();`

Reply:
addressed by implementing an AnalyticsProvider around the `/why-base` page and using `logEventWithContext` in this `ButtonWithLinkAndEventLogging` component